packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_v2.dart

```diff
@@ -0,0 +1,369 @@
+  const LoggingTableV2({super.key, required this.model});
+
+  final LoggingTableModel model;
+
+  @override
+  State<LoggingTableV2> createState() => _LoggingTableV2State();
+}
+
+class _LoggingTableV2State extends State<LoggingTableV2> {
+  late final ScrollController _verticalController = ScrollController();
+  final Set<int> selections = <int>{};
+  final Map<int, double> cachedOffets = {};
+  final normalTextStyle = const TextStyle(color: Colors.black, fontSize: 14.0);
+  String lastSearch = '';
+  final metadataTextStyle = const TextStyle(
+    color: Colors.black,
+    fontStyle: FontStyle.italic,
+    fontSize: 12.0,
+  );
+  double maxWidth = 0.0;
+
+  @override
+  void initState() {
+    super.initState();
+    // On web, disable the browser's context menu since this example uses a custom
+    // Flutter-rendered context menu.
+    if (kIsWeb) {
+      unawaited(BrowserContextMenu.disableContextMenu());
+    }
+  }
+
+  @override
+  void dispose() {
+    if (kIsWeb) {
+      unawaited(BrowserContextMenu.enableContextMenu());
+    }
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Row(
+          children: [
+            Expanded(
+              child: TextField(
+                decoration: const InputDecoration(
+                  border: OutlineInputBorder(),
+                  hintText: 'Filter',
+                ),
+                onSubmitted: (value) {},
+              ),
+            ),
+            const Expanded(
+              child: TextField(
+                decoration: InputDecoration(
+                  border: OutlineInputBorder(),
+                  hintText: 'Search',
+                ),
+              ),
+            ),
+          ],
+        ),
+        Expanded(
+          child: _LoggingTableContextMenu(
+            child: _LoggingTableContents(
+              model: widget.model,
+              verticalController: _verticalController,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _LoggingTableContents extends StatefulWidget {
+  _LoggingTableContents({
+    required this.model,
+    required ScrollController verticalController,
+  }) : _verticalController = verticalController;
+
+  static const _millisecondsUntilCacheProgressShows = 500;
+  static const _millisecondsUntilCacheProgressHelperShows = 2000;
+
+  final LoggingTableModel model;
+  final ScrollController _verticalController;
+
+  @override
+  State<_LoggingTableContents> createState() => _LoggingTableContentsState();
+}
+
+class _LoggingTableContentsState extends State<_LoggingTableContents> {
+  final Stopwatch _progressStopwatch = Stopwatch();
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        widget.model.tableWidth = constraints.maxWidth;
+        _progressStopwatch.reset();
+        return ListenableBuilder(
```

Comment:
use a ValueListenableBuilder so that we don't have to define a local variable for the value